components/datetime/src/any/datetime.rs

```diff
@@ -506,24 +524,17 @@ where {
 }
 
 #[cfg(test)]
-#[cfg(feature = "serde")]
 mod tests {
     use core::str::FromStr;
     use icu::calendar::{AnyCalendar, DateTime};
     use icu::datetime::{options::length, DateTimeFormatter};
     use icu::locid::Locale;
-    use icu_provider::BufferProvider;
 
-    fn test_format(
-        provider: &impl BufferProvider,
-        datetime: &DateTime<AnyCalendar>,
-        locale: &str,
-        expected: &str,
-    ) {
+    fn test_format(datetime: &DateTime<AnyCalendar>, locale: &str, expected: &str) {
         let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Short);
 
-        let dtf = DateTimeFormatter::try_new_with_buffer_provider(
-            provider,
+        let dtf = DateTimeFormatter::try_new_unstable(
+            &icu_testdata::unstable(),
```

Comment:
Question: I thought we couldn't use `icu_testdata::unstable()` in unit tests; why does it work here?

Reply:
beats me